### PR TITLE
add count of indices in cluster

### PR DESCRIPTION
**Pull request by j0hnsmith**
Realised that `/_all/_stats` contains _all_ indices in the cluster, ref #131.

### CODE REVIEW

**Comment by dominikschulz** [comment]
Thanks for the pull request!

IMHO you shouldn't pull in the cluster name. It can change during the runtime of the exporter if e.g. a different node/configuration is made available at the same destination address.

**Comment by j0hnsmith** [comment]
Yeah, I thought about that too. Docs say to change the cluster name you should perform a full cluster restart, I'm _assuming_ (quite a big assumption) that will include the exporter. Changing the cluster name should be a very rare event, it seemed better than an extra http request on every scrape. What do you think? Any alternative ideas?

**Comment by zwopir** [comment]
Hi @j0hnsmith,

I just talked to Dominik and agree with him regarding his comment. For now my idea would be to just not include the cluster name as a constant label. I definitely don't like the idea getting the cluster name on every scrape. On the other hand outdated cluster name information in the labels are more harmful than not having them at all.

There's an issue (#135) also addressing the "cluster name as label" topic. So re-adding the label can be done while working on #135.

Maybe a label updater go routine that regularly gets the labels might be an option to solve the problem. Seems a bit to much logic involved for a simple exporter, thought (i.e. getting the updates in the collectors). What do you think, @dominikschulz ?

**Comment by j0hnsmith** [comment]
Many people run multiple ElasticSearch clusters, in this case metrics without a cluster label will be plain wrong. Feels like it's better to resolve #135 before moving forward with this.

**Comment by dominikschulz** [comment]
IMHO the exported metrics should not include the cluster name at all.
You can and should add that during relabeling.

**Comment by j0hnsmith** [comment]
Relabeling requires additional config, I don't understand how/why that improves the situation, can you elaborate?

**Comment by dannyk81** [comment]
Quite agree with @j0hnsmith here, including this information during relabeling forces us to have some way to sync the Cluster name we define in Elasticsearch and Prometheus configuration, this is very tricky and having this identifier (i.e. Cluster name) provided from the application via the exporter is akin to providing the index names, I do hope you will reconsider your stance.

File: main.go [hunk 1]
```diff
@@ -92,6 +110,10 @@ func main() {
 	}
 }
 
+type bodyData struct {
```
Comment (metalmatze) [inline main.go]: You can put this type into the func itself right before it is used. It seems to be the only place where you use it.

Something like:
```go
bodyData := &struct {
	ClusterName string `json:"cluster_name"`
}{}
```

**Comment by maxromanovsky** [comment]
Nice PR! How about adding also the health of index? Maybe as simple as green/red/yellow?

**Comment by caarlos0** [comment]
count of indices can be obtained with `count by (instance) (elasticsearch_indices_docs_primary)`

**Comment by gavk34** [comment]
@j0hnsmith Can we possibly add the index name as a label to a metric, that way users can simply use Prometheus to sum up that value to give them the count of indices and also the index name - granted we are running a fairly lightweight cluster so not too sure of the cardinality of this.

**Comment by j0hnsmith** [comment]
@gavk34 sounds like you want per index metrics eg name, size, doc count, I think you're looking for the `es.indices` flag.

**Comment by gavk34** [comment]
Ah brilliant @j0hnsmith, we have `es.all` which I falsely believed did everything 🤦‍♂️ 

My bad I should have read the documentation - sorry for the noise 👍

**Comment by zwopir** [comment]
obsoleted by #176

**Pull request closed without merging**